lib/ash_json_api/json_schema/open_api.ex

```diff
@@ -0,0 +1,956 @@
+      include_parameter(),
+      fields_parameter(resource)
+    ] ++
+      read_argument_parameters(route, resource)
+  end
+
+  defp query_parameters(%{type: type}, _resource)
+       when type in [:post_to_relationship, :patch_relationship, :delete_from_relationship] do
+    []
+  end
+
+  defp query_parameters(%{type: type} = route, resource) when type in [:get, :related] do
+    [include_parameter(), fields_parameter(resource)] ++ read_argument_parameters(route, resource)
+  end
+
+  defp query_parameters(_route, resource) do
+    [include_parameter(), fields_parameter(resource)]
+  end
+
+  @spec filter_parameter(resource :: module) :: Parameter.t()
+  defp filter_parameter(resource) do
+    %Parameter{
+      name: :filter,
+      in: :query,
+      description:
+        "Filters the query to results with attributes matching the given filter object",
+      required: false,
+      style: :deepObject,
+      schema: filter_schema(resource)
+    }
+  end
+
+  @spec sort_parameter(resource :: module) :: Parameter.t()
+  defp sort_parameter(resource) do
+    sorts =
+      resource
+      |> Ash.Resource.Info.public_attributes()
+      |> Enum.flat_map(fn attr -> [to_string(attr.name), "-#{attr.name}"] end)
+
+    %Parameter{
+      name: :sort,
+      in: :query,
+      description: "Sort order to apply to the results",
+      required: false,
+      style: :form,
+      explode: false,
+      schema: %Schema{
+        type: :array,
+        items: %Schema{
+          type: :string,
+          enum: sorts
+        }
+      }
+    }
+  end
+
+  @spec page_parameter() :: Parameter.t()
+  defp page_parameter() do
+    %Parameter{
+      name: :page,
+      in: :query,
+      description: "Paginates the response with the limit and offset",
+      required: false,
+      style: :deepObject,
+      schema: %Schema{
+        type: :object,
+        properties: %{
+          limit: %Schema{type: :integer, minimum: 1},
+          offset: %Schema{type: :integer, minimum: 0}
+        }
+      }
+    }
+  end
+
+  @spec include_parameter() :: Parameter.t()
+  defp include_parameter do
+    %Parameter{
+      name: :include,
+      in: :query,
+      required: false,
+      description: "Relationship paths to include in the response",
+      style: :form,
+      explode: false,
+      schema: %Schema{
+        type: :array,
+        items: %Schema{
+          type: :string,
+          pattern: ~r/^[a-zA-Z_]\w*(\.[a-zA-Z_]\w*)*$/
+        }
+      }
+    }
+  end
+
+  @spec filter_parameter(resource :: module) :: Parameter.t()
+  defp fields_parameter(resource) do
+    type = AshJsonApi.Resource.Info.type(resource)
+
+    %Parameter{
+      name: :fields,
+      in: :query,
+      description: "Limits the response fields to only those listed for each type",
+      required: false,
+      style: :deepObject,
+      schema: %Schema{
+        type: :object,
+        additionalProperties: true,
+        properties: %{
+          :"#{type}" => %Schema{
```

Comment:
Technically this has to have a key for the root type and any includable type to be complete.